### DISCUSSION TITLE
Allow changing the font sizes for the rest of the ui.

### DISCUSSION THREAD

**Discussion by chabad360** [GitSquared/edex-ui#742] [Ideas]
**Perceived idea awesomeness**
- [ ] Wow, you don't have this yet???
- [x] Would be super cool.
- [ ] I'm not sure a lot of people would use it, but I'd love to have it.
- [ ] Kind of a gadget, honestly, but why not?

**Perceived idea difficulty**
- [ ] Quit your job first.
- [ ] Would need some work, but nothing impossible.
- [x] Shouldn't be too hard to make.
- [ ] Like, *I* could do it.

---

Some fonts (Roboto Mono for instance) are a tad to big for the UI causing some portions of the UI to overflow, being able to set the font sizes for the different UI components (`h1`, etc) within the theme settings would be awesome.

**Comment by abelurii** [comment]
Traducir a plis español

**Comment by GitSquared** [comment]
You can use the injectCSS property in themes to achieve just that.

**Comment by chabad360** [comment]
yes, but I'd need to set it for each class, which is very inconvenient. I'm wondering if would be possible to have a global value?

**Comment by chabad360** [comment]
If this wont convince you, I don't know what will:

```json
{
    "injectCSS": "section#filesystem{left: 0;width:100vw} section#filesystem>h3.title, section#filesystem>div{width:100vw} section#keyboard{display:none;} div#fs_disp_container>*>h3, section#filesystem.list-view>div#fs_disp_container:not(.disks)>div>h4{font-size:1.2vh;} div#fs_space_bar>h1{font-size:1.4vh;} div#fs_space_bar>h3{font-size:1.3vh;} div#mod_clock>h1{font-size:3.9;} div#mod_clock.mod_clock_twelve>h1>span:last-child{font-size:1.4vh;} div#mod_conninfo>h1{font-size:1.38vh;} div#mod_conninfo>h2{font-size:1.3vh;} div#mod_conninfo>i{font-size:1.1vh;} div#mod_cpuinfo_innercontainer>h1:first-child{font-size: 1.38vh;} div#mod_cpuinfo_innercontainer>h1:first-child>i{font-size: 1.1vh;} div#mod_cpuinfo>i{font-size: 1.2vh;} ul#fuzzyFinder-results>li{font-size:1.4vh;} div#mod_globe>h1{font-size:1.38vh;} div#mod_globe>h2{font-size: 1.3vh;} div#mod_globe>i{font-size: 1.1vh;} div#mod_hardwareInspector_inner>div>*{font-size: 1.2vh;} div#mod_netstat{font-size:1.011vh;} div#mod_netstat_inner>h1:first-child{font-size: 1.38vh;} div#mod_netstat_inner>h1:first-child>i{font-size: 1.1vh;} div#mod_ramwatcher{font-size:1.011vh;} div#mod_ramwatcher_inner>h1:first-child{font-size: 1.38vh;} div#mod_ramwatcher_inner>h1:first-child>i{font-size: 1.1vh;} div#mod_ramwatcher_swapcontainer>h1{font-size: 1.2vh;} h3#mod_ramwatcher_swaptext{font-size: 1.2vh;} div#mod_sysinfo{font-size:1.011vh;} div#mod_toplist>h1{font-size: 1.38vh;} div#mod_toplist>i{font-size:1.1vh;} table#mod_toplist_table>td{font-size: 1.40vh;} div.modal_popup>h1>i{font-size: 2.2vh;} div.modal_popup>h5{font-size: 1.9vh;} div.modal_popup>button{font-size:1.75vh;} div.modal_popup>table>th{font-size: 1.9vh;} div.modal_popup>table>td{font-size: 1.6vh;} div.modal_popup>td>input{font-size:1.3vh;} div.modal_popup>td>select{font-size:1.3vh;} div.modal_popup>summary{font-size: 1.8vh;} h6#settingsEditorStatus{font-size:1.4vh}"
}
```

This is impossible to work with, if I decide to use a different font and I want to change any of the font sizes, I need to go through all of the values here and many of them are the same! It's ridiculous!

**Comment by GitSquared** [comment]
OK, yeah, this is a pain to work with - worse than i thought.
We'd need something like a font size ratio you could set up in the settings and it would multiply the sizes by this ratio in the CSS files.

Unfortunately i'm not working on this much anymore so unless someone opens a PR i can't guarantee a fix coming soon.

Will keep the issue opened.
